File: src/test/scala/com/infotarget/codility/scala/BinaryGapTest.scala
package com.infotarget.codility.scala

import org.scalatest.FunSuite

class BinaryGapTest extends FunSuite {

  test("testSolution") {
    assert(2 === BinaryGap.solution(9))
    assert(4 === BinaryGap.solution(529))
    assert(5 === BinaryGap.solution(1041))
    assert(0 === BinaryGap.solution(15))
    assert(0 === BinaryGap.solution(8))
  }
}

File: src/test/java/com/infotarget/codility/java/BinaryGapTest.java
package com.infotarget.codility.java;

import org.junit.Assert;
import org.junit.Test;

import java.util.function.ToIntFunction;

public class BinaryGapTest {

    private BinaryGap binaryGap = new BinaryGap();

    @Test
    public void testSolution1() throws Exception {
        testSolution(binaryGap::solution1);
    }

    @Test
    public void testSolution2() throws Exception {
        testSolution(binaryGap::solution2);
    }

    private void testSolution(ToIntFunction<Integer> solution) {
        Assert.assertEquals(2, solution.applyAsInt(9));
        Assert.assertEquals(4, solution.applyAsInt(529));
        Assert.assertEquals(5, solution.applyAsInt(1041));
        Assert.assertEquals(0, solution.applyAsInt(15));
        Assert.assertEquals(0, solution.applyAsInt(8));

    }
}